bentoml/cli/bento.py

```diff
@@ -237,3 +245,112 @@ def retrieve(bento, target_dir):
         safe_retrieve(bento_service_bundle_path, target_dir)
 
         click.echo('Service %s artifact directory => %s' % (name, target_dir))
+
+    @cli.command(
+        help='Containerize given Bento into a Docker image',
+        short_help="Containerize given Bento into a Docker image",
+    )
+    @click.argument(
+        "bento", type=click.STRING, callback=parse_bento_tag_callback,
+    )
+    @click.option('--push', is_flag=True)
+    @click.option(
+        '--docker-repository',
+        help="Prepends specified Docker repository to image name.",
+    )
+    @click.option(
+        '-u', '--username', type=click.STRING, required=False,
+    )
+    @click.option(
+        '-p', '--password', type=click.STRING, required=False,
+    )
+    def containerize(bento, push, docker_repository, username, password):
+        """Containerize specified BentoService.
+
+        BENTO is the target BentoService to be containerized, referenced by its name
+        and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+        `bentoml containerize` command also supports the use of the `latest` tag
+        and will automatically use the last built version of your Bento.
+
+        You can also optionally provide a `--push` flag, which will push the built
+        image to the Docker repository specified by the `--docker-repository`.
+
+        If you would like to push to Docker Hub, `--docker-repository` can just be
+        your Docker Hub username. Otherwise, the tag should include the hostname as
+        well. For example, for Google Container Registry, the `--docker-repository`
+        would be `[HOSTNAME]/[PROJECT-ID]`
+        """
+        name, version = bento.split(':')
```

Comment:
Nitpick: this should live with the other docker-related utility functions in `cli/utils.py`. Rationale: even though this is really simple logic, if some other piece of code wants to split up Docker images in `name:tag` format, there should be a single function that does that work.

Reply:
Another nice thing about a single function: if I just provide the name without a tag, that function could provide a reasonable default tag (e.g. "latest"), raise an exception, or do some other sensible thing.

Reply:
Looks like a lot of other CLI functions do similar logic so I might make another issue and PR to refactor that later. Will keep it as is in current PR though

Reply:
One thing we want to have a good handle on is the special tag `latest`.  

Right now, you can use it to get the latest BentoService, for example `,bentoml get IrisClassifier:latest`.  For this command,  I think we should be the image and also make sure to tag the image with both the `latest` tag and the actual version tag.  If a user decided to push to a docker registry, we should push both tags to the registry as well.

This way, the user can use the `latest` tag in their application and still have reference to the previous version of Docker image, if the `latest` tag had been overwritten.

Reply:
@flosincapite great suggestion, and indeed we should clean up lots of other places in the codebase that does this

Reply:
@yubozhao docker registry manages `latest` automatically, I don't think you need to tag or push to `latest` tag separately